src/utils.ts

```diff
@@ -587,34 +587,62 @@ export async function getDelegatesBySpace(
   const spaceIn = ['', space];
   if (space.includes('.eth')) spaceIn.push(space.replace('.eth', ''));
 
+  type Delegation = {
+    delegator: string;
+    delegate: string;
+    space: string;
+    timestamp: number;
+  };
   const PAGE_SIZE = 1000;
-  let result = [];
-  let page = 0;
+  let result: Delegation[] = [];
   const params: any = {
     delegations: {
       __args: {
         where: {
           space_in: spaceIn
         },
         first: PAGE_SIZE,
-        skip: 0
+        skip: 0,
+        orderBy: 'timestamp',
+        orderDirection: 'asc'
       },
       delegator: true,
       space: true,
-      delegate: true
+      delegate: true,
+      timestamp: true
     }
   };
   if (snapshot !== 'latest') {
     params.delegations.__args.block = { number: snapshot };
   }
 
+  const isSameDelegate = (a: Delegation, b: Delegation) => {
+    return (
+      a.delegator === b.delegator &&
+      a.delegate === b.delegate &&
+      a.space === b.space
+    );
+  };
```

Comment:
Should be outside of this function

Reply:
Maybe we should just extract this whole logic inside a utils/delegation.ts